<span style="color:red;font-style:normal;font-family:Georgia">1. IMPORTAMOS LAS LIBRERIAS.</span>

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import chi2_contingency

<span style="color:red;font-style:normal;font-family:Georgia">2. IMPORTAMOS LA DATA DE NUESTRA MUESTRA.</span>

In [58]:
data=pd.read_csv("https://raw.githubusercontent.com/VictorGuevaraP/Estadistica_con_Python/master/Ejemplo_med.csv",sep=";")
data.head()

NEncuesta  Edad  Puntuacion  Duracion    Sexo     Ingreso       Puntcat
0          1    58           5        14  Hombre    Urgencia    Satisfecho
1          2    10           7         4   Mujer  Programado    Satisfecho
2          3    11           2        15   Mujer  Programado  Insatisfecho
3          4    35           9         3  Hombre    Urgencia    Satisfecho
4          5    25           9         2   Mujer    Urgencia    Satisfecho

<span style="color:red;font-style:normal;font-family:Georgia">3. REALIZAMOS UNA TABLA DE CONTINGENCIA PARA VER SI ATENDERSE POR PROGRAMADO O URGENCIA TIENE ALGUN EFECTO SOBRE LA CALIFICACION DE LA ATENCION(SATISFECHO O INSATISFECHO).</span>

In [72]:
observados=pd.crosstab(index=data["Ingreso"],columns=data["Puntcat"],margins=True)
observados.columns=["Insatisfecho_obs","Satisfecho_obs","Total"]
observados.index=["Programado","Urgencia","Total"]
observados

Insatisfecho_obs  Satisfecho_obs  Total
Programado                 7               7     14
Urgencia                   6              10     16
Total                     13              17     30

<span style="color:red;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>

- Aparentemente existe relación entre la variable Ingreso y Puntcat.
- Las personas se encuentran mas satisfechos cuando entran por Urgencia, y Insatisfechos cuando entran por Programado.

<span style="color:red;font-style:normal;font-family:Georgia">4. PARA ESTO APLICAMOS UNA PRUEBA ESTADISTICA, CHI2, CON UN NIVEL DE SIGNIFICANCIA DEL 5% ESTO PARA VER SI HAY DIFERENCIA O NO ENTRE ATENDERSE POR PROGRAMADO O URGENCIA Y QUE ESTO SE CUMPLA TAMBIÉN PARA NUESTRA POBLACIÓN.</span>

![](https://www.medwave.cl/medios//medwave/Diciembre2011/MBE04/Formula%201.jpg)

<span style="color:red;font-style:normal;font-family:Georgia">4.1 EMPEZAMOS FORMULANDO LA HIPOTESIS.</span>

<span style="color:black;font-style:normal;font-family:Georgia">HIPOTESIS :</span>
- H0 :  P1 = P2
- H1 :  P1 ≠ P2

<span style="color:red;font-style:normal;font-family:Georgia">4.2. APLICAREMOS LA PRUEBA CHI2, DONDE SE DEBE CUMPLIR QUE MENOS DEL 20% DE LAS CELDAS DEBEN TENER UNA FRECUENCIA ESPERADA MENOR A 5. PARA ESTO TENEMOS QUE HALLAR LA FRECUENCIA ESPERADA DE CADA PROPORCION.</span>

In [70]:
esperados=pd.DataFrame(test[3])
esperados.columns=["Insatisfecho_esp","Satisfecho_esp","Total"]
esperados.index=["Programado","Urgencia","Total"]
esperados

Insatisfecho_esp  Satisfecho_esp  Total
Programado          6.066667        7.933333   14.0
Urgencia            6.933333        9.066667   16.0
Total              13.000000       17.000000   30.0

<span style="color:red;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>
- Vemos que efectivamente se cumple la condición ya que ninguna proporcion tiene  valor esperado menor a 5, entonces podemos usar chi2.

<span style="color:red;font-style:normal;font-family:Georgia">4.3. POR COMODIDAD PASAREMOS TODAS LAS PROPORCIONES, TANTO LAS OBSERVADAS COMO LAS ESPERADAS, EN TERMINOS RELATIVOS.</span>

In [73]:
observados_rel=round(observados.apply(lambda r: r/len(data)*100,axis=1),2)
observados_rel

Insatisfecho_obs  Satisfecho_obs   Total
Programado             23.33           23.33   46.67
Urgencia               20.00           33.33   53.33
Total                  43.33           56.67  100.00

In [74]:
esperados_rel=round(esperados.apply(lambda r: r/len(data)*100,axis=1),2)
esperados_rel

Insatisfecho_esp  Satisfecho_esp   Total
Programado             20.22           26.44   46.67
Urgencia               23.11           30.22   53.33
Total                  43.33           56.67  100.00

<span style="color:red;font-style:normal;font-family:Georgia">4.4. USANDO CHI2.</span>

In [66]:
test=chi2_contingency(observados)
test

(0.4751131221719459,
 0.9758780582018594,
 4,
 array([[ 6.06666667,  7.93333333, 14.        ],
        [ 6.93333333,  9.06666667, 16.        ],
        [13.        , 17.        , 30.        ]]))

In [75]:
test[1]

0.9758780582018594

<span style="color:red;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>
- Como 0.9759>0.05 quiere decir que no es significativo, no rechazamos H0.
- Entonces a un 95% de confianza en la población todas proporciones son iguales.
- Aunque en la muestra hayan pequeñas diferencias, en la poblacion no podemos estar seguros que haya diferencias entre las proporciones.